src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -306,6 +306,10 @@ public void execute(BuildResultAdapter buildResult) {
                     addCustomValueAndSearchLink(develocity, "CI run", value));
                 envVariable("GITHUB_HEAD_REF", providers).filter(value -> !value.isEmpty()).ifPresent(value ->
                     buildScan.value("PR branch", value));
+                envVariable("GITHUB_ACTION", providers).ifPresent(value ->
```

Comment:
Nitpick, but I'd group all of the env vars together that produce a value + link. In other words, I'd have the GITHUB_HEAD_REF logic last in this code block.